index.js

```diff
@@ -409,13 +409,31 @@ let rcl = {
   createMessageObject(type) {
     return this.createMessage(type).toPlainObject();
   },
+
+  /**
+   * Removes the default signal handler installed by rclnodejs. After calling this, rclnodejs
+   * will no longer clean itself up when a SIGINT is received, it is the application's
+   * responsibility to properly shut down all nodes and contexts.
+   *
+   * Application which wishes to implement its own signal handler logic should call this.
+   * @returns {undefined}
+   */
+  removeSignalHandlers() {
+    process.removeListener('SIGINT', _sigHandler);
+  },
 };
 
-process.on('SIGINT', () => {
-  debug('Caught SIGINT/ctrl+c event and will cleanup and terminate.');
-  rcl.shutdown();
-  process.exit(0);
-});
+const _sigHandler = () => {
+  // shuts down all live contexts. Applications which wishes to use their own signal handlers
+  // should call `rclnodejs.removeSignalHandlers`.
+  debug('Catch ctrl+c event and will cleanup and terminate.');
+  for (let ctx of rcl._contextToNodeArrayMap.keys()) {
+    if (!rcl.isShutdown(ctx)) {
+      rcl.shutdown(ctx);
+    }
```

Comment:
```suggestion
    rcl.shutdown(ctx);
```

This check is already done within `rcl.shutdown()` since #721.